src/module/mixer.rs

```diff
@@ -1,54 +1,61 @@
-use mixlab_protocol::{Mixer4chParams, LineType};
+use mixlab_protocol::{MixerParams, LineType};
 
 use crate::engine::Sample;
 use crate::module::ModuleT;
 
 #[derive(Debug)]
-pub struct Mixer4ch {
-    params: Mixer4chParams,
+pub struct Mixer {
+    params: MixerParams,
+    inputs: Vec<LineType>,
 }
 
-impl ModuleT for Mixer4ch {
-    type Params = Mixer4chParams;
+impl ModuleT for Mixer {
+    type Params = MixerParams;
     type Indication = ();
 
     fn create(params: Self::Params) -> (Self, Self::Indication) {
-        (Mixer4ch { params }, ())
+        let mixer = Mixer {
+            inputs: params.channels.iter().map(|_| LineType::Stereo ).collect(),
+            params,
+        };
+
+        (mixer, ())
     }
 
     fn params(&self) -> Self::Params {
         self.params.clone()
     }
 
     fn update(&mut self, params: Self::Params) -> Option<Self::Indication> {
+        self.inputs = params.channels.iter().map(|_| LineType::Stereo ).collect();
         self.params = params;
         None
     }
 
     fn run_tick(&mut self, _t: u64, inputs: &[Option<&[Sample]>], outputs: &mut [&mut [Sample]]) -> Option<Self::Indication> {
+        const MASTER: usize = 0;
+        const CUE: usize = 1;
+
         let len = outputs[0].len();
 
-        let mut channel_gain: [f64; 4] = [0.0; 4];
+        let mut channel_gain = vec![0.0; self.params.channels.len()];
```

Comment:
Can we move this allocation into the state struct?

Reply:
Fixed in e33fa07